tenders/common/elf.c

```diff
@@ -194,7 +195,8 @@ static int align_up(Elf64_Addr addr, Elf64_Xword align, Elf64_Addr *out_result)
 }
 
 void elf_load(int bin_fd, const char *bin_name, uint8_t *mem, size_t mem_size,
-       uint64_t p_min_loadaddr, uint64_t *p_entry, uint64_t *p_end)
+        uint64_t p_min_loadaddr, guest_mprotect_fn_t t_guest_mprotect, void *t,
```

Comment:
1) Hmm, would you be okay with changing the name to be something a bit more specific, like `void *tender_handle`
 or `*t_guest_mprotect_handle` whatever that is easier to recognize at the call site of `t_guest_mprotect()`?

2) Later on this will be casted to one of:
    ```c
    struct hvt *hvt = t;
    struct spt *spt = t;
    ```
    I wonder if it would make sense with a target-specific pointer type so we avoid having `void *` casts passed around?

Reply:
1. I can do `void *t_guest_mprotect_arg`.
2. Will reply separately re function pointers.